heron/tmaster/src/cpp/manager/stateful-controller.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2015 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __TMASTER_STATEFUL_HELPER_H_
+#define __TMASTER_STATEFUL_HELPER_H_
+
+#include <set>
+#include <string>
+#include "network/network.h"
+#include "manager/tmaster.h"
+#include "basics/basics.h"
+
+namespace heron {
+namespace common {
+class MetricsMgrSt;
+class MultiCountMetric;
+}
+}  // namespace heron
+
+namespace heron {
+namespace tmaster {
+
+class StatefulRestorer;
+class StatefulCheckpointer;
+
+class StatefulController {
```

Comment:
please add comments describing the controller and what it's general area of responsibility is.